backend/app/services/llm/providers/base.py

```
@@ -5,9 +5,99 @@
+
+    expected_type = rule["type"]
+    label = rule["label"]
+
+    if not isinstance(resolved_input, expected_type):
+        actual_label = INPUT_TYPE_LABEL.get(
+            type(resolved_input), type(resolved_input).__name__
+        )
+        hint = (
+            " Please set completion type to 'multimodal' when sending mixed input types."
+            if isinstance(resolved_input, MultiModalInput)
+            else f" Please ensure the input type matches the completion type."
```

Ruff (0.15.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)